node-js-basics-101/node_modules/ws/lib/BufferUtil.js

```diff
@@ -0,0 +1,71 @@
+/*!
+ * ws: a node.js websocket client
+ * Copyright(c) 2011 Einar Otto Stangvik <einaros@gmail.com>
+ * MIT Licensed
+ */
+
+'use strict';
+
+const safeBuffer = require('safe-buffer');
+
+const Buffer = safeBuffer.Buffer;
+
+/**
+ * Merges an array of buffers into a new buffer.
+ *
+ * @param {Buffer[]} list The array of buffers to concat
+ * @param {Number} totalLength The total length of buffers in the list
+ * @return {Buffer} The resulting buffer
+ * @public
+ */
+const concat = (list, totalLength) => {
+  const target = Buffer.allocUnsafe(totalLength);
+  var offset = 0;
+
+  for (var i = 0; i < list.length; i++) {
+    const buf = list[i];
+    buf.copy(target, offset);
+    offset += buf.length;
+  }
+
+  return target;
+};
+
+try {
+  const bufferUtil = require('bufferutil');
+
+  module.exports = Object.assign({ concat }, bufferUtil.BufferUtil || bufferUtil);
```

Comment:
__[jshint]__
* 'object short notation' is available in ES6 (use 'esversion: 6') or Mozilla JS extensions (use moz).
* 'module' is not defined.

<!-- Comment by [SideCI](https://sideci.com) -->